packages/api/src/external/carequality/document/shared.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  DocumentReference,
+  OutboundDocumentQueryResp,
+  OutboundDocumentRetrievalResp,
+  DocumentReference as IHEGWDocumentReference,
+} from "@metriport/ihe-gateway-sdk";
+import {
+  DocumentReference as DocumentReferenceFHIR,
+  DocumentReferenceContent,
+} from "@medplum/fhirtypes";
+import { toFHIRSubject } from "@metriport/core/external/fhir/patient/index";
+import { cqExtension } from "../../carequality/extension";
+import { DocumentReferenceWithId, createDocReferenceContent } from "../../fhir/document";
+import { metriportDataSourceExtension } from "../../fhir/shared/extensions/metriport";
+
+export type DocumentReferenceWithMetriportId = DocumentReference & {
+  id: string;
+};
+
+type IHEResults = OutboundDocumentQueryResp | OutboundDocumentRetrievalResp;
+
+// Create a single array of all the document references from all the document query results
+export function combineDocRefs(documentQueryResults: IHEResults[]): DocumentReference[] {
+  return documentQueryResults.reduce((acc: DocumentReference[], curr) => {
+    const documentReferences = curr.documentReference ?? [];
+    const documentReferencesWithUrl = documentReferences.map(docRef => {
+      return {
+        ...docRef,
+        url: curr.gateway.url,
+      };
+    });
+
+    return [...acc, ...documentReferencesWithUrl];
+  }, []);
+}
+
+export const cqToFHIR = (
+  docId: string,
+  doc: IHEGWDocumentReference,
+  patientId: string,
+  fhirDocRef?: DocumentReferenceFHIR
+): DocumentReferenceWithId => {
+  const baseAttachment = {
+    ...(doc.fileName ? { fileName: doc.fileName } : {}),
+    ...(doc.contentType ? { contentType: doc.contentType } : {}),
+    ...(doc.size ? { size: doc.size } : {}),
+    ...(doc.creation ? { creation: doc.creation } : {}),
+  };
+
+  return {
+    ...(fhirDocRef ? { ...fhirDocRef } : {}),
+    ...(!fhirDocRef ? { description: doc.title ?? undefined } : {}),
+    id: docId,
+    resourceType: "DocumentReference",
+    masterIdentifier: {
+      system: doc.homeCommunityId,
+      value: doc.docUniqueId,
```

Comment:
Shouldn't this be `docId` (Metriport's doc ref id) instead?

We could use `identifier` to store the original ID.